### DISCUSSION TITLE
Both Capacity and Time windows Solver

### DISCUSSION THREAD

**Discussion by walterwootz** [TimefoldAI/timefold-solver#457] [Ideas]
I just discovered this awesome tool and I'm starting to use it in logistic domain, nevertheless, in many logistic routing problems both capacity and time windows are crucial constraints.

Do you have (or plan to) a use case that mixes both capacity and time windows solver? Is there a way to do it using the library?

Thank you!

**Comment by triceo** [comment]
Thanks for your interest in Timefold! You should be able to get what you need by combining these two quickstarts:

https://github.com/TimefoldAI/timefold-quickstarts/tree/stable/use-cases/vehicle-routing-capacity
https://github.com/TimefoldAI/timefold-quickstarts/tree/stable/use-cases/vehicle-routing-time-windows

We may decide to merge them one day, but there are no short-term plans to do it.

**Reply by walterwootz** [reply]
Ok thank you for the answer, I would appreciate a quick instruction on how to accomplish that, one day.

**Reply by ge0ffrey** [reply]
Step 1: Take a look at the code
Step 2: Add the capacity constraint to the time windows quickstart

The only way to accomplish it, is to give it a try :)

**Reply by walterwootz** [reply]
I was able to achieve the goal successfully with just a few steps, following the documentation. Really simple. Many thanks!
